### PR TITLE
CLN/PERF: remove used functions; use C skip list for rolling median

### PR DESCRIPTION
**Pull request by kawochen**
removes some unused code
reverts this commit https://github.com/pydata/pandas/commit/a40226e5c4c3d29e5e2383433c5f0b94c3c0ecc1 

performance consideration

```
import pandas
import numpy
arr = numpy.random.rand(1000000)
%timeit pandas.rolling_median(arr, 1000)
```

master

```
1 loops, best of 3: 4.94 s per loop
```

branch

```
1 loops, best of 3: 821 ms per loop
```

### CODE REVIEW

File: doc/source/whatsnew/v0.17.1.txt [hunk 1]
```diff
@@ -58,7 +58,7 @@ Performance Improvements
 
 
 - Release the GIL on most datetime field operations (e.g. ``DatetimeIndex.year``, ``Series.dt.year``), normalization, and conversion to and from ``Period``, ``DatetimeIndex.to_period`` and ``PeriodIndex.to_timestamp`` (:issue:`11263`)
-
+- ``rolling_median`` uses c skip list implementation
```
Comment (jreback) [inline doc/source/whatsnew/v0.17.1.txt]: add the pr number (as the issue number)

File: pandas/src/skiplist.h [hunk 2]
```diff
@@ -68,33 +69,38 @@ static PANDAS_INLINE int int_min(int a, int b) {
...
 static PANDAS_INLINE void node_decref(node_t *node) {
-  node->ref_count -= 1;
+  --(node->ref_count);
 }
 
 static void node_destroy(node_t *node) {
   int i;
   if (node) {
-    if (node->ref_count == 1) {
+    if (node->ref_count <= 1) {
```
Comment (jreback) [inline pandas/src/skiplist.h]: are sure about this?

**Comment by jreback** [comment]
you *may* be able to insert ``nogil`` (at the top-level), and make almost all of ``skiplist.pyx`` ``nogil`` (obviously where you raise you have re-acquire).

File: pandas/src/skiplist.h (hunk 2 above)
Comment (kawochen) [inline pandas/src/skiplist.h]: I think it makes clean-up code cleaner.  I will write something to test for leaks a bit later.

**Comment by kawochen** [comment]
```
In [2]: import pandas
import numpy
arr = numpy.random.rand(1000000)

In [4]: from pandas.util import testing

In [5]: @testing.test_parallel(2)
def g():
    pandas.rolling_median(arr, 1000)
   ...:     

In [6]: @testing.test_parallel(1)
def f():
    pandas.rolling_median(arr, 1000)


In [7]: %timeit f()
1 loops, best of 3: 752 ms per loop


In [8]: %timeit g()
1 loops, best of 3: 972 ms per loop
```

**Comment by jreback** [comment]
can you add an asv benchmark? not sure we have much for rolling in general......

File: doc/source/whatsnew/v0.17.1.txt [hunk 3]
```diff
@@ -59,7 +59,7 @@ Performance Improvements
 
 
 - Release the GIL on most datetime field operations (e.g. ``DatetimeIndex.year``, ``Series.dt.year``), normalization, and conversion to and from ``Period``, ``DatetimeIndex.to_period`` and ``PeriodIndex.to_timestamp`` (:issue:`11263`)
-
+- ``rolling_median`` uses c skip list implementation (:issue:`11450`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.17.1.txt]: say ``Improved performance of ....``

**Comment by kawochen** [comment]
OK.  might as well release the GIL on roll_*

**Comment by jreback** [comment]
yep - same or another

**Comment by kawochen** [comment]
Added asv benchmarks for the gil release.  But I can't get any of those to show in asv bench at all.  Have I written the tests incorrectly?
I do ```asv continuous upstream/master HEAD -b gil`` while on my branch.

A simple `timeit` does show the improvement, e.g.
for `rolling_mean`
```
import pandas
import numpy
from pandas.util.testing import test_parallel
arr = numpy.random.rand(1000000)
@test_parallel(num_threads=2)
def f():
     pandas.rolling_mean(arr, 100)
%timeit f()
```
branch
```
100 loops, best of 3: 12.5 ms per loop
```
master 
```
10 loops, best of 3: 19.5 ms per loop
```
for `rolling_kurt`
branch 
```
10 loops, best of 3: 65.5 ms per loop
```
master
```
10 loops, best of 3: 98.8 ms per loop
```

**Comment by jreback** [comment]
thanks!

**Pull request merged**